File: backend/profile-service/src/main/java/com/musa/profile/dto/UpdateProfileRequest.java
package com.musa.profile.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record UpdateProfileRequest (

        @Size(max = 500, message = "Biography must not exceed 500 characters")
        String biography,

        @NotNull(message = "Privacy value is required")
        Boolean privateProfile
){
}

File: frontend/lib/features/profile/data/models/update_profile_request.dart
class UpdateProfileRequest{
    final String? biography;
    final bool privateProfile;

    const UpdateProfileRequest({
        this.biography,
        required this.privateProfile,
    });

    Map<String, dynamic> toJson(){
        return {
            'biography': biography,
            'privateProfile': privateProfile,
        };
    }
}